### PR TITLE
KKP User Cluster Management with Terrafrom Rest API and Cluster CRDs

### PR DESCRIPTION
**Pull request by toschneck**
closes https://github.com/kubermatic/sig-cs-infra/issues/62

**What this PR does / why we need it**:

Shows the ability how KKP and Terraform could interact. The Provider is considered as example to extend it to the custom needs and cloud providers. As first example kubevirt is added

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
- Adding KKP REST API Terraform provider (Scope: Example Implementation to extend to custom needs)
- Adding Examples How to manage KKP User Cluster with Cluster CRDs at the seed level
```

### CODE REVIEW

File: components/api/cluster-management-by-crds/README.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+```
+4. Ensure Parameters are matching. Somehow you need to ensure matching values of:
+   * Project ID ``: Secrets, Labels, MachineDeployments
+   * Cloud Provider Credentials and Specs
+     * vsphere: folder path
+
+
+## Clustertemplate Management
+
+Another option is to manage the [`ClusterTempalte`](https://docs.kubermatic.com/kubermatic/v2.23/references/crds/#clustertemplate) object. Therefore, a non initialized template get created and separate instance object creates a copy of it. **BUT** any change to the clustertempalte will **NOT** get applied to the instance.
```
Comment (dharapvj) [inline components/api/cluster-management-by-crds/README.md:87]:
```suggestion
Another option is to manage the [`ClusterTemplate`](https://docs.kubermatic.com/kubermatic/v2.23/references/crds/#clustertemplate) object. Therefore, a non initialized template get created and separate instance object creates a copy of it. **BUT** any change to the clustertempalte will **NOT** get applied to the instance.
```

**Comment by dharapvj** [comment]
May we should add few more things via `kubectl` :

1. Addon and AddonConfig CR
2. ApplicationInstallation CR
3. User-MLA Alerts (No CR I guess. Gets applied directly in Cortex?)
4. User-MLA AlertManager config (AlertManager CR)

**Comment by toschneck** [comment]
> May we should add few more things via `kubectl` :
> 
> 1. Addon and AddonConfig CR
> 2. ApplicationInstallation CR
> 3. User-MLA Alerts (No CR I guess. Gets applied directly in Cortex?)
> 4. User-MLA AlertManager config (AlertManager CR)
> 
> I can try to add these after your PR is merged

aggree created new issue for this https://github.com/kubermatic/community-components/issues/123

**Comment by toschneck** [comment]
fixed issues and add architecture overview.

**Pull request merged**